src/app/clusters/commodity-tariff-server/commodity-tariff-server.cpp

```diff
@@ -0,0 +1,172 @@
+    case DayPatterns::Id:
+        return aEncoder.Encode(mDelegate.GetDayPatterns());
+    case CalendarPeriods::Id:
+        return aEncoder.Encode(mDelegate.GetCalendarPeriods());
+    case IndividualDays::Id:
+        return aEncoder.Encode(mDelegate.GetIndividualDays());
+    case CurrentDay::Id:
+        return aEncoder.Encode(mDelegate.GetCurrentDay());
+    case NextDay::Id:
+        return aEncoder.Encode(mDelegate.GetNextDay());
+    case CurrentDayEntry::Id:
+        return aEncoder.Encode(mDelegate.GetCurrentDayEntry());
+    case CurrentDayEntryDate::Id:
+        return aEncoder.Encode(mDelegate.GetCurrentDayEntryDate());
+    case NextDayEntry::Id:
+        return aEncoder.Encode(mDelegate.GetNextDayEntry());
+    case NextDayEntryDate::Id:
+        return aEncoder.Encode(mDelegate.GetNextDayEntryDate());
+    case TariffComponents::Id:
+        return aEncoder.Encode(mDelegate.GetTariffComponents());
+    case TariffPeriods::Id:
+        return aEncoder.Encode(mDelegate.GetTariffPeriods());
+    case CurrentTariffComponents::Id:
+        return aEncoder.Encode(mDelegate.GetCurrentTariffComponents());
+    case NextTariffComponents::Id:
+        return aEncoder.Encode(mDelegate.GetNextTariffComponents());
+    case DefaultRandomizationOffset::Id:
+        if (!HasFeature(Feature::kRandomization))
+        {
+            return CHIP_IM_GLOBAL_STATUS(UnsupportedAttribute);
+        }
+        return aEncoder.Encode(mDelegate.GetDefaultRandomizationOffset());
+    case DefaultRandomizationType::Id:
+        if (!HasFeature(Feature::kRandomization))
+        {
+            return CHIP_IM_GLOBAL_STATUS(UnsupportedAttribute);
+        }
+        return aEncoder.Encode(mDelegate.GetDefaultRandomizationType());
+
+    /* FeatureMap - is held locally */
+    case FeatureMap::Id:
+        return aEncoder.Encode(mFeature);
+    }
+
+    /* Allow all other unhandled attributes to fall through to Ember */
+    return CHIP_NO_ERROR;
+}
+
+void Instance::InvokeCommand(HandlerContext & handlerContext)
+{
+    using namespace Commands;
+
+    switch (handlerContext.mRequestPath.mCommandId)
+    {
+    case GetTariffComponent::Id:
+        HandleCommand<GetTariffComponent::DecodableType>(
+            handlerContext, [this](HandlerContext & ctx, const auto & commandData) { HandleGetTariffComponent(ctx, commandData); });
+        return;        
+    case GetDayEntry::Id:
+        HandleCommand<GetDayEntry::DecodableType>(
+            handlerContext, [this](HandlerContext & ctx, const auto & commandData) { HandleGetDayEntry(ctx, commandData); });
+        return;
+    }
+}
+
+void Instance::HandleGetTariffComponent(HandlerContext & ctx, const Commands::GetTariffComponent::DecodableType & commandData)
+{
+    Commands::GetTariffComponentResponse::Type response;
+
+    Status status = mDelegate.GetTariffComponentInfoById(commandData.tariffComponentID, 
+                                                         response.label,
```

Comment:
How does the callee know when it's safe to deallocate the buffers backing the label and list here?  And the label of the `tariffComponent`?